src/CodeGenerator/FactoryGenerator.php

```diff
@@ -7,24 +7,38 @@
 
 namespace Zend\Di\CodeGenerator;
 
-use Psr\Container\ContainerInterface;
-use Zend\Code\Generator\ClassGenerator;
-use Zend\Code\Generator\DocBlockGenerator;
-use Zend\Code\Generator\FileGenerator;
-use Zend\Code\Generator\MethodGenerator;
-use Zend\Code\Generator\ParameterGenerator;
+use SplFileObject;
 use Zend\Di\ConfigInterface;
-use Zend\Di\Resolver\AbstractInjection;
+use Zend\Di\Exception\RuntimeException;
 use Zend\Di\Resolver\DependencyResolverInterface;
+use Zend\Di\Resolver\InjectionInterface;
 use Zend\Di\Resolver\TypeInjection;
 
+use function file_get_contents;
+use function strrpos;
+use function strtr;
+use function substr;
+
 /**
  * Generates factory classes
  */
 class FactoryGenerator
 {
     use GeneratorTrait;
 
+    private const TEMPLATE_FILE = __DIR__ . '/../../templates/factory.template';
+    private const PARAMETERS_TEMPLATE = <<< '__CODE__'
+        if (empty($options)) {
+            $args = [
+                %s
+            ];
+        } else {
+            $args = [
+                %s
+            ];
+        }
```

Comment:
I'm wondering if you could simplify this with a ternary?

```php
$args = empty($options)
    ? [
        %s
    ]
    : [
        %s
    ];
```

That said... now that I've written the above, these are the same. I'm assuming the two `%s` placeholders get filled with something different? If so, the ternary will be more clear in the generated code.